webalyzer/collector/static/collect.js

```diff
@@ -42,47 +53,81 @@
     };
     req.send();
   }
+
   function postCSS(url, callback) {
     _getCSS(url, function(status, response) {
       if (status === 200) {
-        var req = request();
+        var source_hash = makeHash(response);
+
+        _postHash(source_hash, 'css', domain, function(status) {
+          if(status === 200) {
+            console.log('We have already collected CSS matching ' + source_hash);
+          } else {
+            console.log('Sending CSS.');
+            var req = request();
+            var data = new FormData();
+            data.append('url', url);
+            data.append('domain', domain);
+            data.append('css', response);
+            data.append('source_hash', source_hash);
+            _post(data, function(status) {
+              if (status === 200) {
+                console.log("We have already collected this CSS");
+              if (callback) callback(url, false);
+              } else if (status === 201) {
+                console.log("Yum! Collected another chunk of CSS");
+              if (callback) callback(url, true);
```

Comment:
I know it's petty but the indentation here really is hard to read. I keep reading the `if` statements as being on the same level, when in fact they're nested.